// ng-src/ng-src.es5.js
import { Directive, EventEmitter, Input, NgModule, Output, ViewContainerRef } from '@angular/core';
var NgSrcDirective = (function () {
    /**
     * @param {?} view
     */
    function NgSrcDirective(view) {
        this.view = view;
        this.ngOnLoad = new EventEmitter();
    }
    Object.defineProperty(NgSrcDirective.prototype, "hostImage", {
        /**
         * @return {?}
         */
        get: function () {
            return this.view.element.nativeElement;
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @return {?}
     */
    NgSrcDirective.prototype.ngOnChanges = function () {
        this.fetchImage(this.ngSrc);
    };
    /**
     * @param {?} url
     * @return {?}
     */
    NgSrcDirective.prototype.fetchImage = function (url) {
        var /** @type {?} */ httpRequest = new XMLHttpRequest();
        httpRequest.open('GET', url, true);
        httpRequest.responseType = 'arraybuffer';
        httpRequest.onload = this.onRequestLoaded.bind(this);
        httpRequest.send();
    };
    /**
     * @param {?} ev
     * @return {?}
     */
    NgSrcDirective.prototype.onRequestLoaded = function (ev) {
        var /** @type {?} */ request = (ev.target);
        var /** @type {?} */ imageUrl = this.createLocalResource(request);
        this.setImageSource(imageUrl);
        this.ngOnLoad.emit(request);
    };
    /**
     * @param {?} request
     * @return {?}
     */
    NgSrcDirective.prototype.createLocalResource = function (request) {
        var /** @type {?} */ arrayBufferView = new Uint8Array(request.response);
        var /** @type {?} */ blob = new Blob([arrayBufferView], { type: this.getContentType(request) });
        return URL.createObjectURL(blob);
    };
    /**
     * @param {?} url
     * @return {?}
     */
    NgSrcDirective.prototype.setImageSource = function (url) {
        this.hostImage.src = url;
    };
    /**
     * @param {?} request
     * @return {?}
     */
    NgSrcDirective.prototype.getContentType = function (request) {
        return request.getResponseHeader('content-type');
    };
    return NgSrcDirective;
}());
NgSrcDirective.decorators = [
    { type: Directive, args: [{
                selector: '[ngSrc]img'
            },] },
];
/**
 * @nocollapse
 */
NgSrcDirective.ctorParameters = function () { return [
    { type: ViewContainerRef, },
]; };
NgSrcDirective.propDecorators = {
    'ngSrc': [{ type: Input },],
    'ngOnLoad': [{ type: Output },],
};
var NgSrcModule = (function () {
    function NgSrcModule() {
    }
    return NgSrcModule;
}());
NgSrcModule.decorators = [
    { type: NgModule, args: [{
                declarations: [NgSrcDirective],
                exports: [NgSrcDirective],
            },] },
];
/**
 * @nocollapse
 */
NgSrcModule.ctorParameters = function () { return []; };
/**
 * Generated bundle index. Do not edit.
 */
export { NgSrcDirective, NgSrcModule };
